lightning-invoice/src/payment.rs

```diff
@@ -173,10 +173,32 @@ where
 	logger: L,
 	event_handler: E,
 	/// Caches the overall attempts at making a payment, which is updated prior to retrying.
-	payment_cache: Mutex<HashMap<PaymentHash, usize>>,
+	payment_cache: Mutex<HashMap<PaymentHash, PaymentAttempts>>,
 	retry_attempts: RetryAttempts,
 }
 
+/// Storing minimal payment attempts information required for determining if a outbound payment can
+/// be retried.
+#[derive(Debug,Clone,Copy)]
+struct PaymentAttempts {
+    /// This should be >= 1 as we only insert PaymentAttempts for a PaymentHash if there is at least
+    /// one attempt.
+    counts: usize,
+    latest_attempted_at: SystemTime,
```

Comment:
Good catch. It's not used anywhere now. But it will be useful for improving logging? Should I remove it and use it for improving logging msg?

https://github.com/lightningdevkit/rust-lightning/pull/1418/files#diff-5165a75ba263a549c7fb9f4b555e449e44b0b680a80a9715a99d11157802a0d7R452

Reply:
IMO, I would remove it if it's not being used.